server/plugin.go

```diff
@@ -74,37 +86,107 @@ func (p *Plugin) OnActivate() error {
 	}
 	p.BotUserID = botID
 
-	channel, appErr := p.API.GetChannelByName(team.Id, channelSplit, false)
+	p.API.LogInfo("bot user", "id", p.BotUserID)
+
+	// get or create channel and return mattermost channel id
+	err = p.getOrCreateMattermostChannels(teamChannels)
+	if err != nil {
+		return err
+	}
```

Comment:
The comment says "return mattermost channel id", but we only return an error. We should either change the comment and method name to make it obvious we're mutating something, or instead return a new structure. I would prefer returning a new array, even if we're mutating it internally. Same with `resolveAndSetTeamIDs`